### PR TITLE
Add jenkins_scripts for Docker cloud, OpenStack Cloud, GitHub Oauth

### PR DESCRIPTION
**Pull request by leel8on**
This adds a jenkins_script that configures a docker cloud. It also avoids the problem of creating duplicate entries when run multiple time which was a problem I had with the original Ansible setup. 

This Works:tm: but I wasn't sure if we wanted to pull out specific settings and make them attributes, or go about this some other way. Not sure if this something that we want to be able to reuse and should be put in another recipe and then just set the attributes in the powerci recipe.

Thoughts?

### CODE REVIEW

File: recipes/powerci.rb [hunk 1]
```diff
@@ -88,3 +88,112 @@
 )
 
 include_recipe 'osl-jenkins::master'
+
+# Jenkins needs to be restarted before running the following script otherwise
+# the imports fail
+jenkins_command 'safe-restart'
```
Comment (ramereth) [inline recipes/powerci.rb]: Why does it need to be restarted?

Comment (leel8on) [inline recipes/powerci.rb]: This is the output from a converge without restarting Jenkins.

```
STDOUT: 
       STDERR: ERROR: Unexpected exception occurred while performing groovy command.
       org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
       RemoteClass: 7: unable to resolve class com.nirima.jenkins.plugins.docker.DockerTemplate
        @ line 7, column 1.
          import com.nirima.jenkins.plugins.docker.DockerTemplate
          ^
       
       RemoteClass: 9: unable to resolve class com.nirima.jenkins.plugins.docker.launcher.DockerComputerSSHLauncher
        @ line 9, column 1.
          import com.nirima.jenkins.plugins.docker.launcher.DockerComputerSSHLauncher
          ^
       
       RemoteClass: 10: unable to resolve class hudson.plugins.sshslaves.SSHConnector
        @ line 10, column 1.
          import hudson.plugins.sshslaves.SSHConnector
          ^
       
       RemoteClass: 6: unable to resolve class com.nirima.jenkins.plugins.docker.DockerCloud
        @ line 6, column 1.
          import com.nirima.jenkins.plugins.docker.DockerCloud
          ^
       
       RemoteClass: 8: unable to resolve class com.nirima.jenkins.plugins.docker.DockerTemplateBase
        @ line 8, column 1.
          import com.nirima.jenkins.plugins.docker.DockerTemplateBase
          ^
       
       5 errors
       
```
Maybe I just need to add those plugins to `['osl-jenkins']['restart_plugins']` ?

Comment (ramereth) [inline recipes/powerci.rb]: Yup!

**Comment by leel8on** [comment]
@ramereth Added the docker-plugin and its dependencies to the restart_plugins. Seems to work now without the extra restart.

**Comment by leel8on** [comment]
[Jenkins/Plugin Configuration](https://trello.com/c/yQ8Q8gQy/449-jenkins-plugin-configuration)

**Comment by leel8on** [comment]
@ramereth 

I'm removing the WIP label from this. There are some still things that needed to be worked on, but I think merging this in its current state would be a good idea to give @DesertBeagle a base to work off of.

**Comment by ramereth** [comment]
@leel8on OK, can you please at least fix the ChefSpec errors?

**Comment by leel8on** [comment]
@ramereth ChefSpec errors fixed.

**Comment by ramereth** [comment]
@leel8on looks like you need to also fix ServerSpec

**Comment by leel8on** [comment]
@ramereth ServerSpec now fixed.

**Comment by leel8on** [comment]
Oops. I probably should have bumped instead of merged, but this recipe is not currently running anywhere and will need some modification before it is.

**Pull request merged**